### DISCUSSION TITLE
How can I validate for standalone API

### DISCUSSION THREAD

**Discussion by liam997** [remix-run/remix#7854] [Q&A]
I had create a standalone API like this. It will receive JSON body from other server. 

Sample: 
```
import { json } from "@remix-run/node";

export const action = async ({params, request}) => {

    let raw_data = await request.text();
    let body = JSON.parse(raw_data)

    const customerProfile = JSON.stringify({
        session_id: body?.session_id,
        first_name: body?.first_name,
        last_name: body?.last_name,
        phone: body?.phone,
        email: body?.email,
    })
    // Do something here

    return json(
        { ...result },
        {
            headers: {
                "Access-Control-Allow-Origin": "*",
                "Access-Control-Allow-Methods": "POST"
            },
        }
    );
}
```
My question is: Is there any lib I can validate the data by library like Joi?

**Comment by dimitrisnl** [comment]
The most popular library today is [zod](https://github.com/colinhacks/zod)

Alternatives are [yup](https://github.com/jquense/yup) & [superstruct](https://github.com/ianstormtaylor/superstruct)

Also since action & loader run on the server, you can use `Joi` without issues.
